flake.nix

```diff
@@ -69,6 +70,37 @@
 
           shellHook = ''
             [ -e .git/hooks/pre-commit ] || pre-commit install --install-hooks && pre-commit install --hook-type commit-msg
+
+            PGDATA="./.pg";
+
+            # Initialize a local database if necessary.
+            if [ ! -e $PGDATA ]; then
+              echo -e "\nInitializing PostgreSQL in $PGDATA\n"
+              initdb $PGDATA --no-instructions -A trust
+              if pg_ctl -D $PGDATA start; then
+                cd fission-server
+                diesel database setup --database-url postgres://localhost:5432/fission-server
+                cd ..
```

Comment:
Nitpick, no need to change, but just so you know, you can also use a sub-shell like so:
```bash
(cd fission-server; diesel database setup --database-url postgres://localhost:5432/fission-server)
```
Removing the need to call `cd ..`
More info: https://unix.stackexchange.com/a/366010